packages/rs-dapi-client/src/dapi_client.rs

```diff
@@ -194,81 +251,40 @@ impl DapiRequestExecutor for DapiClient {
-                            })?;
+                let retries = retries_counter.load(std::sync::atomic::Ordering::Acquire);
+
+                let execution_result = result
+                    .map(|inner| {
+                        tracing::trace!(response = ?inner, "received {} response", response_name);
+
+                        ExecutionResponse {
+                            inner,
+                            retries,
+                            address: address.clone(),
                         }
+                    })
+                    .map_err(|inner| {
+                        tracing::debug!(error = ?inner, "received error: {inner}");
 
-                        tracing::trace!(?response, "received {} response", response_name);
-                    }
-                    Err(error) => {
-                        if error.can_retry() {
-                            if applied_settings.ban_failed_address {
-                                let mut address_list = self
-                                    .address_list
-                                    .write()
-                                    .expect("can't get address list for write");
-                                tracing::warn!(
-                                    ?address,
-                                    ?error,
-                                    "received server error, banning address"
-                                );
-                                address_list.ban_address(&address).map_err(|error| {
-                                    ExecutionError {
-                                        inner: DapiClientError::AddressList(error),
-                                        retries: retries_counter
-                                            .load(std::sync::atomic::Ordering::Acquire),
-                                        address: Some(address.clone()),
-                                    }
-                                })?;
-                            } else {
-                                tracing::debug!(
-                                    ?address,
-                                    ?error,
-                                    "received server error, we should ban the node but banning is disabled"
-                                );
-                            }
-                        } else {
-                            tracing::debug!(
-                                ?error,
-                                "received server error, most likely the request is invalid"
-                            );
+                        ExecutionError {
+                            inner,
+                            retries,
+                            address: Some(address.clone()),
                         }
-                    }
-                };
+                    });
 
-                let retries = retries_counter.load(std::sync::atomic::Ordering::Acquire);
+                ban_failed_address::<R::Response, DapiClientError>(
```

Comment:
the name is a bit confusing - it looks like the address will be banned after this line. Maybe `ban_address_if_failed` ? or `update_banned_status`? Or sth?

Reply:
update_address_banned_status ?